### PR TITLE
Low duration values now error instead of running tests forever

### LINKED ISSUE #45: issue with duration in milliseconds
I did a `-d 5ms -r 5` and the tests never stopped. Weird.

### PR DESCRIPTION
**Pull request by stezu**
Fixes #45

### CODE REVIEW

File: lib/run.js [hunk 1]
```diff
@@ -32,10 +32,14 @@ function validateOpts(options) {
     
     validateNumberProp(options, 'rate');
     validateNumberProp(options, 'concurrent');
-    
-    // hack to enable concurrency mode,
-    // since forkmaster expects this to be defined
-    options.rate = options.rate || 0;
```
Comment (catdad) [inline lib/run.js]: Did I ever make `formaster` not error if `rate` was undefined?

File: lib/run.js [hunk 2]
```diff
@@ -32,10 +32,14 @@ function validateOpts(options) {
...
+
+    if (options.concurrent) {
+        delete options.rate;
+    }
+
+    if (options.rate < 0 || Math.floor(options.rate * (options.duration / 1000)) < 1) {
+        throw new Error('duration is too low');
```
Comment (catdad) [inline lib/run.js]: Hmm... this technically means that the rate to duration ratio is not doable.

I was kind of considering running a single test in this case. Presumably, the only intention a user could have for calling run (or using the CLI) is to run a test, which means that expecting at least 1 run is pretty reasonable. Right?

File: lib/run.js (hunk 1 above)
Comment (stezu) [inline lib/run.js]: Dunno if it works in all cases, but it worked when I ran concurrent and rate tests. I guess you fixed it?

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "test": "mocha -R spec",
     "coverage": "istanbul cover --dir coverage node_modules/mocha/bin/_mocha",
-    "grandma": "node lib/cli.js"
+    "grandma": "node bin/_cli"
```
Comment (catdad) [inline package.json]: Eh. This is a convenience script. All `_cli` does is resolve to the locally installed `grandma`'s `cli.js` file. So frankly, I don't really care about this one. I actually just installed it globally now and don't even use this npm script.

File: lib/run.js (hunk 2 above)
Comment (stezu) [inline lib/run.js]: Is another run mode more reasonable in that case? If you can't get a test to run in your allotted time, maybe it makes more sense to say `grandma run count=1`.

File: lib/run.js (hunk 1 above)
Comment (catdad) [inline lib/run.js]: I hate it when I do that and don't tell myself. I'll have to check it out.

GitKraken won't work on my home computer (same white screen and crash as if I were on a VPN), and I am too lazy to walk to my laptop.

File: package.json (hunk 3 above)
Comment (stezu) [inline package.json]: I just changed it since `lib/cli.js` isn't a real thing.

File: lib/run.js (hunk 2 above)
Comment (catdad) [inline lib/run.js]: This is really the fault of my weird fraction system to figure out how many tests to run when. When the fraction converges to 0, that doesn't necessarily mean I should run 0 tests.

Most similar systems, I think, actually start a test and then attempt to figure out if/when they should start another one. I was just too lazy to read Go code to figure it out.

Am I crazy?

File: package.json (hunk 3 above)
Comment (catdad) [inline package.json]: All the more showing that I don't use it anymore. I'll allow it.

**Pull request merged**